tests/scalers/loki/loki_test.go

```diff
@@ -0,0 +1,252 @@
+//go:build e2e
+// +build e2e
+
+package loki_test
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/joho/godotenv"
+	"github.com/stretchr/testify/assert"
+	"k8s.io/client-go/kubernetes"
+
+	. "github.com/kedacore/keda/v2/tests/helper"
+)
+
+// Load environment variables from .env file
+var _ = godotenv.Load("../../.env")
+
+const (
+	testName = "loki-test"
+)
+
+var (
+	testNamespace         = fmt.Sprintf("%s-ns", testName)
+	deploymentName        = fmt.Sprintf("%s-deployment", testName)
+	monitoredAppName      = fmt.Sprintf("%s-monitored-app", testName)
+	publishDeploymentName = fmt.Sprintf("%s-publish", testName)
+	scaledObjectName      = fmt.Sprintf("%s-so", testName)
+	lokiServerName        = fmt.Sprintf("%s-server", testName)
+	minReplicaCount       = 0
+	maxReplicaCount       = 2
+)
+
+type templateData struct {
+	TestNamespace         string
+	DeploymentName        string
+	MonitoredAppName      string
+	PublishDeploymentName string
+	ScaledObjectName      string
+	LokiServerName        string
+	BackTick              string
+	MinReplicaCount       int
+	MaxReplicaCount       int
+}
+
+const (
+	deploymentTemplate = `apiVersion: apps/v1
+kind: Deployment
+metadata:
+  labels:
+    app: test-app
+  name: {{.DeploymentName}}
+  namespace: {{.TestNamespace}}
+spec:
+  replicas: 0
+  selector:
+    matchLabels:
+      app: test-app
+  template:
+    metadata:
+      labels:
+        app: test-app
+        type: keda-testing
+    spec:
+      containers:
+      - name: nginx
+        image: nginx:1.14.2
+        ports:
+        - containerPort: 80
+---
+`
+
+	monitoredAppDeploymentTemplate = `apiVersion: apps/v1
+kind: Deployment
+metadata:
+  labels:
+    app: {{.MonitoredAppName}}
+  name: {{.MonitoredAppName}}
+  namespace: {{.TestNamespace}}
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: {{.MonitoredAppName}}
+  template:
+    metadata:
+      labels:
+        app: {{.MonitoredAppName}}
+        type: {{.MonitoredAppName}}
+    spec:
+      containers:
+      - name: nginx
+        image: nginx:1.14.2
+        ports:
+        - containerPort: 80
+---
+`
+
+	scaledObjectTemplate = `apiVersion: keda.sh/v1alpha1
+kind: ScaledObject
+metadata:
+  name: {{.ScaledObjectName}}
+  namespace: {{.TestNamespace}}
+spec:
+  scaleTargetRef:
+    name: {{.DeploymentName}}
+  minReplicaCount: {{.MinReplicaCount}}
+  maxReplicaCount: {{.MaxReplicaCount}}
+  pollingInterval: 3
+  cooldownPeriod:  1
+  triggers:
+  - type: loki
+    metadata:
+      serverAddress: http://loki.{{.TestNamespace}}.svc.cluster.local:3100
+      threshold: '0.5'
+      query: sum(rate({namespace="test"} |= {{.BackTick}}keda-scaler{{.BackTick}} [1m]))
```

Comment:
```suggestion
      query: sum(rate({namespace="{{.TestNamespace}}"} |= {{.BackTick}}keda-scaler{{.BackTick}} [1m]))
```
Here is the problem with e2e tests, the query is wrong xD we should use the test namespace